src/world.rs

```diff
@@ -210,8 +225,33 @@ impl World {
 
     pub fn apply_action(&mut self, action: &Action) {
         match action {
-            Action::Destroy { at } => self.destroy_cube(at.clone()),
-            Action::Add { at, block } => self.add_cube(at.clone(), block.clone())
+            Action::Destroy { at } => {
+                let cubes_to_add = self.destroy_cube(at.clone());
+                self.cubes_to_draw.destroy_cube(&at);
+                for cube in cubes_to_add {
+                    self.cubes_to_draw.add_cube(&cube)
+                }
+
+            },
+            Action::Add { at, block } => {
+                let (cubes_to_destroy, cube) = self.add_cube(at.clone(), block.clone());
+                self.cubes_to_draw.add_cube(&cube);
+                for position in cubes_to_destroy {
+                    self.cubes_to_draw.destroy_cube(&position)
+                }
+            }
+        }
+    }
+
+    pub fn apply_action_no_render(&mut self, action: &Action) {
+        match action {
+            Action::Destroy { at } => {
+                self.destroy_cube(at.clone());
+
+            },
+            Action::Add { at, block } => {
+                self.add_cube(at.clone(), block.clone());
+            }
```

Comment:
Remove this function. 

Again the responsability principle.

The caller must NOT KNOW ANYTHING about the inner logic of the class. 

It is the class that knows whether `cubes_to_draw` is set to `None` or not.